servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -431,6 +439,13 @@ private Single<C> selectConnection0(final Predicate<C> selector, @Nullable final
         }
         return establishConnection
                 .flatMap(newCnx -> {
+                    if (forceNewConnectionAndReserve && !newCnx.tryReserve()) {
+                        return failed(StacklessConnectionRejectedException.newInstance(
```

Comment:
This will leak the connection. We have to follow the same pattern as below for other exceptions:
```
return newCnx.closeAsync().concat(failed(...));
```